### PR TITLE
feat: add `component` property to `ComponentContext`

### PR DESCRIPTION
**Pull request by GeomKid**
## Pull Request Type

- [x] Feature addition
- [ ] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description
Reimplementing PR #1268 

## Python Compatibility

- [x] I've ensured my code works on Python `3.10.x`
- [ ] I've ensured my code works on Python `3.11.x`

## Checklist

- [x] I've run the `pre-commit` code linter over all edited files
- [x] I've tested my changes on supported Python versions
- [ ] I've added tests for my code, if applicable
- [x] I've updated / added  documentation, where applicable

### CODE REVIEW

File: interactions/models/internal/context.py [hunk 1]
```diff
@@ -780,6 +780,21 @@ async def edit_origin(
             self.message_id = message.id
             return message
 
+    @property
+    def component(self) -> typing.Optional[typing.Union[Button, BaseSelectMenu]]:
```
Comment (Damego) [inline interactions/models/internal/context.py]: Make it return `BaseComponent`

File: interactions/models/internal/context.py [hunk 2]
```diff
@@ -780,6 +780,21 @@ async def edit_origin(
...
+        """
+        The component that you interacted.
+
+        .. versionadded:: 5.3.0
+        :rtype: Optional[Union[Button, "BaseSelectMenu"]]
```
Comment (Damego) [inline interactions/models/internal/context.py]: We don't use sphinx (readthedocs) anymore

**Review by Damego — CHANGES_REQUESTED** [review]

**Review by Damego — APPROVED** [review]
looks good to me

**Review by Wolfhound905 — CHANGES_REQUESTED** [review]
"The component that you interacted."
 
Please fix this grammatical error. I think you meant:
"The component that was interacted with."

**Comment by GeomKid** [comment]
> The component that was interacted with

fixed :)

**Review by LordOfPolls — APPROVED** [review]

**Pull request merged**